File: apps/api/schemas/private_document.py
"""Private document request/response schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from apps.api.models.enums import (
    PrivateDocumentApprovalStatus,
    PrivateDocumentEntityType,
    PrivateDocumentKind,
)


class PrivateDocumentResponse(BaseModel):
    """Safe document metadata — storage_key is intentionally excluded.

    The frontend must call the /download endpoint to obtain a signed URL.
    """

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    entity_type: PrivateDocumentEntityType
    entity_id: uuid.UUID
    kind: PrivateDocumentKind
    original_filename: str
    mime_type: str
    size_bytes: int
    uploaded_by_user_id: uuid.UUID | None = None
    uploaded_by_platform_user_id: uuid.UUID | None = None
    # Approval workflow fields (migration 0037)
    approval_status: PrivateDocumentApprovalStatus = PrivateDocumentApprovalStatus.APPROVED
    approved_by_user_id: uuid.UUID | None = None
    approved_at: datetime | None = None
    rejected_reason: str | None = None
    created_at: datetime
    updated_at: datetime


class PrivateDocumentListResponse(BaseModel):
    items: list[PrivateDocumentResponse]
    total: int


class DownloadUrlResponse(BaseModel):
    signed_url: str
    expires_in: int


class ApproveDocumentRequest(BaseModel):
    pass  # No body needed; approver identity comes from CurrentUser


class RejectDocumentRequest(BaseModel):
    reason: str  # min 5 chars validated at service layer


class PendingDocumentFilter(BaseModel):
    entity_type: PrivateDocumentEntityType | None = None
    assigned_to_me: bool = False

File: apps/web/src/lib/types/private-document.ts
export type PrivateDocumentKind =
  | "rera_id"
  | "passport"
  | "emirates_id"
  | "trade_license"
  | "noc"
  | "contract"
  | "kyc"
  | "other";

export type PrivateDocumentEntityType = "agent" | "customer" | "deal" | "tenant";

export interface PrivateDocument {
  id: string;
  tenant_id: string;
  entity_type: PrivateDocumentEntityType;
  entity_id: string;
  kind: PrivateDocumentKind;
  storage_key: string;
  original_filename: string;
  mime_type: string;
  size_bytes: number;
  uploaded_by_user_id: string | null;
  uploaded_by_platform_user_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface PrivateDocumentListResponse {
  items: PrivateDocument[];
  total: number;
}

export interface DownloadUrlResponse {
  signed_url: string;
  expires_in: number;
}
